src/handlers/transfers/validator.js

```diff
@@ -85,7 +85,16 @@ const validatePositionAccountByNameAndCurrency = async function (participantName
 const validateDifferentDfsp = (payload) => {
   const isPayerAndPayeeDifferent = (payload.payerFsp !== payload.payeeFsp)
   if (!isPayerAndPayeeDifferent) {
-    reasons.push(`Payer and Payee should be different.`)
+    reasons.push(`Payer and Payee should be different`)
+    return false
+  }
+  return true
+}
+
+const validateFspiopSourceMatchesPayer = (payload, headers) => {
+  const matched = (headers && headers['fspiop-source'] && headers['fspiop-source'] === payload.payerFsp)
```

Comment:
What about validating the payeeFsp for the Fulfils?

Reply:
I agree, but this "hotfix" update is only for the prepare.